### PR TITLE
Fix coverity issues compaction_job, compaction_picker

### PR DESCRIPTION
**Pull request by pdvian**
Summary:
db/compaction_job.cc:
  ReportStartedCompaction(compaction);

CID 1419863 (#1 of 1): Uninitialized scalar field (UNINIT_CTOR)
2. uninit_member: Non-static class member bottommost_level_ is not initialized in this constructor nor in any functions that it calls.

db/compaction_picker_universal.cc:
7struct InputFileInfo {
   	2. uninit_member: Non-static class member level is not initialized in this constructor nor in any functions that it calls.

CID 1405355 (#1 of 1): Uninitialized scalar field (UNINIT_CTOR)
4. uninit_member: Non-static class member index is not initialized in this constructor nor in any functions that it calls.
 38  InputFileInfo() : f(nullptr) {}

db/dbformat.h:
 ParsedInternalKey()
 84      : sequence(kMaxSequenceNumber)  // Make code analyzer happy

CID 1168095 (#1 of 1): Uninitialized scalar field (UNINIT_CTOR)
2. uninit_member: Non-static class member type is not initialized in this constructor nor in any functions that it calls.
 85  {}  // Intentionally left uninitialized (for speed)

### CODE REVIEW

File: db/dbformat.h [hunk 1]
```diff
@@ -81,8 +81,8 @@ struct ParsedInternalKey {
   ValueType type;
 
   ParsedInternalKey()
-      : sequence(kMaxSequenceNumber)  // Make code analyzer happy
-  {}  // Intentionally left uninitialized (for speed)
+      : sequence(kMaxSequenceNumber),  // Make code analyzer happy
+        type(kTypeDeletion) { } // Intentionally left uninitialized (for speed)
```
Comment (yiwu-arbug) [inline db/dbformat.h]: Can you revert the change for dbformat.h? As the comment said it is left uninitialized for speed, although I don't know how much difference there is.

Comment (pdvian) [reply db/dbformat.h]: @yiwu-arbug made the changes as requested. Kindly review the changes.

**Comment by yiwu-arbug** [comment]
Merging.
